app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java

```diff
@@ -0,0 +1,140 @@
+        View rootView = inflater.inflate(R.layout.lab57_activity_fragment, container, false);
+
+        // Get values for spinner
+        String[] valuesForSpinner = getResources().getStringArray(R.array.spinner);
+
+        // List of objects for title and description
+        List<AppInfo> appInfoValues = new ArrayList<AppInfo>();
+
+        PackageManager packageManager = getActivity().getPackageManager();
+        List<ApplicationInfo> packages = packageManager.getInstalledApplications(0);
+
+        for (ApplicationInfo packageInfo : packages) {
+            // Size of app
+            double size = Utils.getSize(packageManager, packageInfo);
+            // Getting icon of app
+            Drawable icon = Utils.getIcon(packageManager, packageInfo);
+            // Adding all values to object
+            AppInfo singleItem = new AppInfo(packageInfo.packageName, packageInfo.targetSdkVersion, size,icon);
+            appInfoValues.add(singleItem);
+        }
+
+        ListView listView = (ListView) rootView.findViewById(R.id.list_view);
+
+        relativeAdapter_ = new AdapterForRelative(this.getActivity(), appInfoValues);
+        linearAdapter_ = new AdapterForLinear(this.getActivity(), appInfoValues);
+
+        listView.setAdapter(linearAdapter_);
+
+        spinnerAdapter_ = new ArrayAdapter<String>(
+                getActivity(),
+                R.layout.support_simple_spinner_dropdown_item,
+                valuesForSpinner
+        );
+
+        menuSpinner_ = (Spinner) rootView.findViewById(R.id.menu_spinner);
+        menuSpinner_.setAdapter(spinnerAdapter_);
+
+        menuSpinner_.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
+                String spinnerMode = parent.getItemAtPosition(pos).toString();
+
+                ListView listView = (ListView) getActivity().findViewById(R.id.list_view);
+
+                Mode layoutMode = Mode.get(spinnerMode);
+
+                switch (layoutMode) {
+                    case LINEAR:
+                        listView.setAdapter(linearAdapter_);
+                        relativeAdapter_.notifyDataSetChanged();
+                        break;
+                    case RELATIVE:
+                        listView.setAdapter(relativeAdapter_);
+                        linearAdapter_.notifyDataSetChanged();
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> arg0) {
+                // Nothing to do
+            }
+        });
+
+        return rootView;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        ListView listView = (ListView) getActivity().findViewById(R.id.list_view);
+        listView.setAdapter(relativeAdapter_);
```

Comment:
Is it good idea to store the mode in SPreferences and restore already chosen one?